File: VB/XtraGauge_Animation/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraGauges.Core.Model
Imports System.Collections.Generic

Namespace XtraGauge_Animation
	Partial Public Class Form1
		Inherits Form
		Private functions As New Dictionary(Of String, IEasingFunction)()
		Private modes As New Dictionary(Of String, EasingMode)()
		Private timer As Timer
		Private random As New Random(DateTime.Now.Millisecond)

		Public Sub New()
			InitializeComponent()
			InitData()
			AddHandler Load, AddressOf Form1_Load
		End Sub

		Private Sub InitData()
			functions.Add("BackEase", New BackEase())
			functions.Add("ElasticEase", New ElasticEase())
			functions.Add("PowerEase", New PowerEase(1))
			functions.Add("QuadraticEase", New QuadraticEase())
			functions.Add("QuinticEase", New QuinticEase())
			functions.Add("SineEase", New SineEase())
			functions.Add("ExponentialEase", New ExponentialEase())
			functions.Add("CircleEase", New CircleEase())
			functions.Add("BounceEase", New BounceEase())
			functions.Add("CubicEase", New CubicEase())
			modes.Add("EaseIn", EasingMode.EaseIn)
			modes.Add("EaseOut", EasingMode.EaseOut)
			modes.Add("EaseInOut", EasingMode.EaseInOut)
			For Each item In modes
				EaseMode.Items.Add(item.Key)
			Next item
			arcScaleComponent1.EnableAnimation = True
		End Sub

		Private Sub ChangeEaseFunction(ByVal sender As Object, ByVal e As EventArgs) Handles BackEase.CheckedChanged, ElasticEase.CheckedChanged, BounceEase.CheckedChanged, PowerEase.CheckedChanged, QuadraticEase.CheckedChanged, CubicEase.CheckedChanged, QuinticEase.CheckedChanged, SineEase.CheckedChanged, ExponentialEase.CheckedChanged, CircleEase.CheckedChanged
			Dim radioButton As RadioButton = TryCast(sender, RadioButton)
			If radioButton.Checked Then
				arcScaleComponent1.EasingFunction = functions((CType(sender, RadioButton)).Name)
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			timer = New Timer()
			timer.Interval = 2000
			AddHandler timer.Tick, AddressOf OnTimerTick
			timer.Start()
		End Sub

		Private Sub OnTimerTick(ByVal sender As Object, ByVal e As EventArgs)
			arcScaleComponent1.Value = AnimateScaleValue(arcScaleComponent1)
		End Sub

		Private Function AnimateScaleValue(ByVal scale As IScale) As Single
			Return random.Next(0, 100)
		End Function

		Private Sub EaseMode_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles EaseMode.SelectedIndexChanged
			arcScaleComponent1.EasingMode = modes(EaseMode.SelectedItem.ToString())
		End Sub

	End Class
End Namespace

File: CS/XtraGauge_Animation/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraGauges.Core.Model;
using System.Collections.Generic;

namespace XtraGauge_Animation {
    public partial class Form1 : Form {
        Dictionary<string, IEasingFunction> functions = new Dictionary<string, IEasingFunction>();
        Dictionary<string, EasingMode> modes = new Dictionary<string, EasingMode>();
        Timer timer;
        Random random = new Random(DateTime.Now.Millisecond);

        public Form1() {
            InitializeComponent();
            InitData();
            this.Load += new EventHandler(Form1_Load);
        }

        private void InitData() {
            functions.Add("BackEase", new BackEase());
            functions.Add("ElasticEase", new ElasticEase());
            functions.Add("PowerEase", new PowerEase(1));
            functions.Add("QuadraticEase", new QuadraticEase());
            functions.Add("QuinticEase", new QuinticEase());
            functions.Add("SineEase", new SineEase());
            functions.Add("ExponentialEase", new ExponentialEase());
            functions.Add("CircleEase", new CircleEase());
            functions.Add("BounceEase", new BounceEase());
            functions.Add("CubicEase", new CubicEase());
            modes.Add("EaseIn", EasingMode.EaseIn);
            modes.Add("EaseOut", EasingMode.EaseOut);
            modes.Add("EaseInOut", EasingMode.EaseInOut);
            foreach (var item in modes) {
                EaseMode.Items.Add(item.Key);
            }
            arcScaleComponent1.EnableAnimation = true;
        }

        private void ChangeEaseFunction(object sender, EventArgs e) {
            RadioButton radioButton = sender as RadioButton;
            if (radioButton.Checked) {
                arcScaleComponent1.EasingFunction = functions[((RadioButton)sender).Name];
            }
        }

        void Form1_Load(object sender, EventArgs e) {
            timer = new Timer();
            timer.Interval = 2000;
            timer.Tick += new EventHandler(OnTimerTick);
            timer.Start();
        }

        void OnTimerTick(object sender, EventArgs e) {
            arcScaleComponent1.Value = AnimateScaleValue(arcScaleComponent1);
        }

        float AnimateScaleValue(IScale scale) {
            return random.Next(0, 100);
        }

        private void EaseMode_SelectedIndexChanged(object sender, EventArgs e) {
            arcScaleComponent1.EasingMode = modes[EaseMode.SelectedItem.ToString()];
        }

    }
}
